datumaro/plugins/pointcloud_format/converter.py

```diff
@@ -0,0 +1,302 @@
+        image_info = OrderedDict([("id", str(index)), ])
+        if isinstance(item.pcd, str) and osp.isfile(item.pcd):
+            path = item.pcd.replace(os.sep, '/')
+            filename= path.rsplit("/", maxsplit=1)[-1]
+        else:
+            filename = self._context._make_pcd_filename(item)
+        image_info["name"] = filename
+        PointCloudPath.WRITE_FILES.append({index: filename})
+
+        if item.has_pcd:
+            if self._context._save_images:
+                self._context._image_dir = osp.join(self._context._default_dir, "pointcloud")
+
+                related_dir = osp.join(self._context._default_dir, "related_images")
+                related_dir = osp.join(related_dir, f"{filename.rsplit('.', maxsplit=1)[0]}_pcd" )
+
+                self._context._save_pcd(item,
+                                          osp.join(self._context._image_dir, filename))
+
+                for i, rimage in enumerate(item.related_images):
+
+                    try:
+                        name = rimage["name"]
+                    except AttributeError:
+                        name = "".join(random.choice(string.ascii_lowercase) for i in range(6))
+                        name = f"{name}.jpg"
+
+                    path = osp.join(related_dir, name)
+
+                    path = osp.abspath(path)
+                    os.makedirs(osp.dirname(path), exist_ok=True)
+
+                    if isinstance(rimage["image"], ByteImage):
+                        with open(path, 'wb') as f:
+                            f.write(item.get_bytes())
+                    else:
+                        save_image(path, rimage["image"].data)
+
+                    path = osp.join(related_dir, f"{name}.json")
+                    self._image_json["name"] = name
+                    with open(path, "w") as f:
+                        json.dump(self._image_json, f, indent=4)
+        else:
+            log.debug("Item '%s' has no image info", item.id)
+
+    def write_key_id_data(self, f):
+        json.dump(self._key_id_data, f, indent=4)
+
+    def write_meta_data(self, f):
+        json.dump(self._meta_data, f, indent=4)
+
+    def write_frame_data(self, f, key):
+        frame = {
+            "description": "",
+            "key": self.get_video_key(int(key)),
+            "tags": [],
+            "objects": self._label_objects,
+            "figures": {}
+        }
+
+        if self._frame_data.get(key):
+            frame["figures"] = self._frame_data[key]
+
+        json.dump(frame, f, indent=4)
+
+
+class PointCloudConverter(Converter):
+    DEFAULT_IMAGE_EXT = PointCloudPath.IMAGE_EXT
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('--reindex', action='store_true',
+            help="Assign new indices to frames (default: %(default)s)")
+        parser.add_argument('--allow-undeclared-attrs', action='store_true',
```

Comment:
I can't see `allow-undeclared-attrs` is used anywhere.

Reply:
Resolved